boards/litex_sim/src/main.rs

```diff
@@ -0,0 +1,415 @@
+//! Board file for a LiteX SoC running in a Verilated simulation
+
+#![no_std]
+// Disable this attribute when documenting, as a workaround for
+// https://github.com/rust-lang/rust/issues/62184.
+#![cfg_attr(not(doc), no_main)]
+#![feature(const_in_array_repeat_expressions)]
+
+use capsules::virtual_alarm::{MuxAlarm, VirtualMuxAlarm};
+use kernel::capabilities;
+use kernel::common::dynamic_deferred_call::{DynamicDeferredCall, DynamicDeferredCallClientState};
+use kernel::common::StaticRef;
+use kernel::component::Component;
+use kernel::hil::time::{Alarm, Timer};
+use kernel::Chip;
+use kernel::InterruptService;
+use kernel::Platform;
+use kernel::{create_capability, debug, static_init};
+use rv32i::csr;
+
+mod io;
+mod litex_generated_constants;
+
+// This module contains the LiteX SoC configuration options, register
+// positions, interrupt mappings and other implementation details of
+// the generated bitstream.
+//
+// Its values are used throughout the file, hence import it under a
+// short name.
+use litex_generated_constants as socc;
+
+/// Structure for dynamic interrupt mapping, depending on the SoC
+/// configuration
+struct LiteXSimInterruptablePeripherals {
+    uart0: &'static litex_vexriscv::uart::LiteXUart<'static, socc::SoCRegisterFmt>,
+    timer0: &'static litex_vexriscv::timer::LiteXTimer<
+        'static,
+        socc::SoCRegisterFmt,
+        socc::ClockFrequency,
+    >,
+    ethmac0: &'static litex_vexriscv::liteeth::LiteEth<'static, socc::SoCRegisterFmt>,
+}
+
+impl InterruptService<()> for LiteXSimInterruptablePeripherals {
```

Comment:
I'm assuming this isn't in chips because it's configurable? Maybe just add a comment saying that.